src/engine/source/api/src/test/handlers.cpp

```diff
@@ -0,0 +1,1139 @@
+#include "api/test/handlers.hpp"
+
+#include <algorithm>
+#include <optional>
+#include <regex>
+#include <string>
+
+#include <fmt/format.h>
+
+#include <eMessages/eMessage.h>
+#include <eMessages/test.pb.h>
+
+#include <api/adapter.hpp>
+#include <api/catalog/resource.hpp>
+#include <json/json.hpp>
+
+namespace
+{
+
+using namespace api::sessionManager;
+
+using api::catalog::Catalog;
+using api::catalog::Resource;
+
+using api::adapter::fromWazuhRequest;
+using api::adapter::genericError;
+using api::adapter::toWazuhResponse;
+
+using router::Router;
+using router::TEST_ROUTE_MAXIMUM_PRIORITY;
+using router::TEST_ROUTE_MINIMUM_PRIORITY;
+
+constexpr uint8_t TEST_DEFAULT_PROTOCOL_QUEUE {1}; ///< Default protocol queue
+constexpr uint8_t TEST_MIN_PROTOCOL_QUEUE {0};     ///< Minimum protocol queue
+constexpr uint8_t TEST_MAX_PROTOCOL_QUEUE {255};   ///< Maximum protocol queue
+
+constexpr auto API_SESSIONS_DATA_FORMAT = R"({"name":"","id":"","creationdate":0,"lifespan":0,"description":"",)"
+                                          R"("filtername":"","policyname":"","routename":""})"; ///< API session data
+                                                                                                ///< format
+
+constexpr auto ASSET_NAME_FIELD_FORMAT = R"("name":"{}")";    ///< JSON name field format, where '{}' is the asset name
+constexpr auto FILTER_CONTENT_FORMAT =
+    R"({{"name": "{}", "check":[{{"TestSessionID":{}}}]}})"; ///< Filter content format
+constexpr auto TEST_FILTER_FULL_NAME_FORMAT = "filter/{}_filter/0"; ///< Filter name format, '{}' is the session name
+constexpr auto TEST_POLICY_FULL_NAME_FORMAT = "policy/{}_policy/0"; ///< Policy name format, '{}' is the session name
+constexpr auto TEST_ROUTE_NAME_FORMAT = "{}_route";                 ///< Route name format, '{}' is the session name
+constexpr auto TEST_FIELD_TO_CHECK_IN_FILTER = "TestSessionID";    ///< Field to check in filter.
+constexpr auto TEST_DEFAULT_PROTOCOL_LOCATION = "api.test";         ///< Default protocol location
+
+constexpr auto FILTER_NOT_REMOVED_MSG = "Filter '{}' could not be removed from the catalog: {}";
+constexpr auto POLICY_NOT_REMOVED_MSG = "Policy '{}' could not be removed from the catalog: {}";
+constexpr auto ROUTE_NOT_REMOVED_MSG = "Route '{}' could not be removed from the router: {}";
+constexpr auto SESSION_NOT_FOUND_MSG = "Session '{}' could not be found";
+constexpr auto SESSION_NOT_REMOVED_MSG = "Session '{}' could not be removed from the sessions manager";
+
+constexpr auto WAZUH_EVENT_FORMAT = "{}:{}:{}"; ///< Wazuh event format
+
+constexpr auto DEFAULT_TIMEOUT {1000};
+
+std::atomic<bool> callbackRunning(false);
+std::mutex queueMutex;
```

Comment:
Lots of semaphores and sync variables, hard to follow. Why is this declared here? Missing documentation.